include/common/get-bool.h

```diff
@@ -4,12 +4,23 @@
 #include <stdbool.h>
 
 /**
- * get_bool - interpret string and return boolean
- * @s: string to interpret
+ * get_bool - return boolean value of string
+ * @string: string to interpret
+ * @ret: boolean variable to set
  *
- * Note: This merely performs a case-insensitive check for 'yes' and 'true'.
- * Returns false by default.
+ * Note: @var will not be touched if the string is not one of 'yes', 'no',
```

Comment:
s/@var/@ret/

Suggest renaming from
- `get_bool()` to `parse_bool()`
- `get_bool_fallback()` to `parse_bool_with_default()`
- `get_bool2()` to `parse_bool_as_int()`.

The last one isn't immediately clear to me though, it could also mean the input is a bool and we cast it to an int. Can't come up with a better name right now.

I am also not sure about the default/fallback variant returning the result vs the other ones supplying a pointer to the result. I have a feeling that all 3 should use the same variant. The docs for the first two also mentions returning the bool.

Reply:
Thank you. Very helpful feedback. 
We now just have one `parse_bool()` to cover all. It now seems a whole lot simpler.
I must have been a bit tired when I came up with all those versions :smile: